pkg/apis/core/validation/validation.go

```diff
@@ -61,8 +61,14 @@ const isInvalidQuotaResource string = `must be a standard resource for quota`
 const fieldImmutableErrorMsg string = apimachineryvalidation.FieldImmutableErrorMsg
 const isNotIntegerErrorMsg string = `must be an integer`
 const isNotPositiveErrorMsg string = `must be greater than zero`
-const csiDriverNameRexpErrMsg string = "must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character"
-const csiDriverNameRexpFmt string = `^[a-zA-Z0-9][-a-zA-Z0-9_.]{0,61}[a-zA-Z-0-9]$`
+
+// The CSI Spec specifies that the driver name starts with an alphanumeric char.
+// https://github.com/container-storage-interface/spec/blob/master/spec.md
+// A DNS1035 label starts with a letter (https://tools.ietf.org/html/rfc1035#section-2.3.1).
+// Also the CSI Spec says the driver name can contain ".". A DNS1035 label does not include ".".
+// For DNS1035, a "." is between a subdomain and a label.
+const csiDriverNameRexpErrMsg string = "must consist of alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character"
+const csiDriverNameRexpFmt string = `^[a-zA-Z0-9][-a-zA-Z0-9.]{0,61}[a-zA-Z0-9]$`
```

Comment:
[dns1123SubdomainFmt](https://sourcegraph.com/github.com/kubernetes/apimachinery@master/-/blob/pkg/util/validation/validation.go#L131) only allows lower case while driver name can be upper or lower case.

We might be able to use [dns1123SubdomainFmt] for topology prefix, but it sounds odd because CSI spec say DNS1035.

Reply:
I think [dns1123SubdomainFmt] won't work for the prefix either because it has dns1123 label before and after the "." while the prefix can have "." any where between the first and last char.

Reply:
I think the intention was to follow dns subdomain syntax as much as possible, although I can see how the csi spec is conflicting.  The first sentence says it should follow the "domain name format":
```
 // The name MUST follow domain name notation format
  // (https://tools.ietf.org/html/rfc1035#section-2.3.1).
```

But then the later sentence does not describe limitations on "." other than it can be anywhere in between:
```
It MUST be 63
  // characters or less, beginning and ending with an alphanumeric
  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
  // alphanumerics between.
```

I would probably go with the more restrictive syntax, which is dns domain.  Maybe we can follow up and clarify the csi spec later.

Reply:
That works.